### ISSUE TITLE
Remove default key mapping

### ISSUE THREAD

**Issue by albohlabs** [NvChad/NvChad#506]
I would like to remove the `copy_whole_file` to `C-a` mapping because i would like to use the default number increment functionality of vim. 
Is there a in-built way to achieve this?

**Comment by macarthy** [comment]
In custom/init.lua,  its something like 


```
map("n", "<C-a>", '<C-U><C-R>=v:count1<CR>Cycle<CR>', opt)
```

**Comment by albohlabs** [comment]
I was thinking to override the mapping too, but would not be my first solution because this list of resets needs to be curated and i'm lazy.
Another thought i had was to set a shortcut for `copy_whole_file` in the `custom/init.lua` file directly 
```
M.mappings = {
   copy_whole_file = "<leader>cf"
}
```
but this introduces only a second mapping.

So no way of doing this without adding another mapping?

**Comment by albohlabs** [comment]
Solved it. The problem was on my side. After looking up where the mapping for `C-a` was defined by using `:verbose map` i saw it was defined in my `~/.local/share/nvim/sessions/...` folder somehow. After deleting the session folder the mapping from `custom/init.lua` worked as expected. 

Btw, thank you for this project. Dropped my rly old vim config in favor of this. :heart:

**Comment by Mate2xo** [comment]
Hi, kind of an old topic but I guess there is no real way to remove a default mapping.
Following [these instructions](https://nvchad.github.io/config/Custom%20config#change-non-plugin-mapping) doesn't actually *changes* a mapping, but just *adds* another one (so I guess there's actually no point of doing this in the `custom/chadrc.lua` ?)

What I ended up doing is using `vim.cmd('unmap xx')` in `custom/chadrc.lua`. E.g.: I don't want `j` to be remapped to `gj`
```lua
M.mappings = {
  misc = function()
    vim.cmd('unmap j')
    vim.cmd('unmap k')
  end
}
```

Did I miss something ? I feel the `misc` is not supposed to be used for that

And thanks for this very interesting config. Very easy to read/understand, and instructive

**Comment by siduck** [comment]
@Mate2xo I will push a re-implementation of mappings today soon

**Comment by Mate2xo** [comment]
Great, thanks!

**Comment by siduck** [comment]
> Great, thanks!

https://github.com/NvChad/NvChad/commit/0844431d37b762c4fa0ff37dba28280848cf3f09

**Comment by lijunzh** [comment]
So, what should we do now after https://github.com/NvChad/NvChad/commit/0844431d37b762c4fa0ff37dba28280848cf3f09 to disable a default mapping. For example, I like the default vim binding for "<S-b>". I can remap "<C-n>" to create a new buffer in `custom/init.lua`. But "<S-b>" still maps to "create a new buffer". How do I force to `umap` a mapping probably in NvChad?

**Comment by lijunzh** [comment]
@siduck, thank a lot. I checked this page before but overlooked these line. I guess I skipped the "Add new mappings" section since I wasn't trying to add new mapping.

**Comment by lijunzh** [comment]
@siduck , the suggested approach will not delete the entry in "WhichKey". Instead, this leave it in the following stage: "B ->   ". The alternative solution i used is to explicitly say what the original binding should be as below:
```lua
M.general  = {
  n = {
    ["B"]  = { "B", "Previous Word." },
  }
}
```
Do you have any suggestion to delete the unmap keybinding in WhichKey.

**Comment by rainbow-bash** [comment]
> Hi, kind of an old topic but I guess there is no real way to remove a default mapping. Following [these instructions](https://nvchad.github.io/config/Custom%20config#change-non-plugin-mapping) doesn't actually _changes_ a mapping, but just _adds_ another one (so I guess there's actually no point of doing this in the `custom/chadrc.lua` ?)
> 
> What I ended up doing is using `vim.cmd('unmap xx')` in `custom/chadrc.lua`. E.g.: I don't want `j` to be remapped to `gj`
> 
> ```lua
> M.mappings = {
>   misc = function()
>     vim.cmd('unmap j')
>     vim.cmd('unmap k')
>   end
> }
> ```
> 
> Did I miss something ? I feel the `misc` is not supposed to be used for that
> 
> And thanks for this very interesting config. Very easy to read/understand, and instructive

DUDE what do  u mean is k and j gj and gk by default? becaus i see a lot of slowness compared to lunarvim

**Comment by siduck** [comment]
k is a lsp mapping -_- @MarcosIgnacioo  ,  idk ab out gj, gk.

**Comment by siduck** [comment]
from nvchad v2.5 you all dont need custom syntax for disabling mappings.

just use unmap or `vim.keymap.del`

**Comment by rainbow-bash** [comment]
> k is a lsp mapping -_- @MarcosIgnacioo , idk ab out gj, gk.

dude im so sorry if i sounded like a dick lol really love you all for making this <3

**Comment by siduck** [comment]
since nvchad no longer uses custom syntax for mapping, it uses `vim.keymap.set` ,  so you can define any syntax you want and just loop over & use that function and use vim.keymap.del to delete the mapping

**Comment by kmanisk** [comment]
to replace a keymap in nvchad you can add this to

1) add this to your nvchad

-- Add custom mappings here

M.mappings = require "custom.mappings"

2) then create a dir call custom > mappings.lua inside the lua dir of nvchad

3) inside mappings.lua

local nomap = vim.keymap.del

nomap("n", "<leader>h")

nomap("n", "<leader>v")

nomap("n", "<leader>x")

or you can add you own keymap which will override and take priority over the nvchad keymaps

map("n", "<S-j>", ":bnext<CR>", { desc = "Next Buffer" })

map("n", "<S-k>", ":bprevious<CR>", { desc = "Previous Buffer" })

this is what i prefer

**Comment by siduck** [comment]
@kmanisk custom dir isnt needed now at all. just put mappings in mappings.lua man

**Comment by Kinetic-Energyy** [comment]
> @kmanisk custom dir isnt needed now at all. just put mappings in mappings.lua man

How to change the default leader x keymap to leader q while removing leader x how to do that

**Comment by siduck** [comment]
> > @kmanisk custom dir isnt needed now at all. just put mappings in mappings.lua man
> 
> How to change the default leader x keymap to leader q while removing leader x how to do that

go to nvchad main repo and see what is leader + x mapped to, then map to <leader> +q

and then use vim.keymap.del to delete leader +x

**Comment by kmanisk** [comment]
> > @kmanisk custom dir isnt needed now at all. just put mappings in mappings.lua man
> 
> How to change the default leader x keymap to leader q while removing leader x how to do that

`-- overrided keymaps by from the base nvchad
map("n", "<leader>q", function()
	require("nvchad.tabufline").close_buffer()
end, { desc = "buffer close" })`

**Comment by Kinetic-Energyy** [comment]
> > > @kmanisk thanks worked

**Comment by w-tr** [comment]
Happy New Year.

I was curious as to why \<C-a\> wasn't working so

TLDR delimiter start
In nvim I ran the command :redir followed by :verbose map.
This did not provide information as to what plugin was highjacking \<C-a\>.

Therefore I went into ~/.local/share/nvim and did a grep \<c-a\>

Returned results were telescope/[builtin/[__git.lua | init.lua] | mappings.lua], lazy...example.lua, which-key...util_spec.lua and nvim-cmp...keymap_spec.lua

I edited example.lua then opened gnome term and saw it inc and dec working. I thought i solved the problem.
So i opened Wezterm and tried ctrl+a and saw it didn't work.
TLDR delimiter end

It turns out people could also be having the problem if they're a twat like I was and use Wezterm with a .wezterm.lua that overwrites ctrl+a to be the wezterm leader key... which takes priority over nvim and nvchad keybindings.

I think there should be a mantra on where to debug first.
Do you know of any general terminal emulator cords that we should steer clear of with regards to nvchad mapping?
